### PR TITLE
refactor(query): unify oct(), hex() and bin()

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- Unify the behavior of the three functions
- Add `write_row()` to `StringColumnBuilder`, replacing `write_and_commit!()`

Part of https://github.com/datafuselabs/databend/issues/7242

### CODE REVIEW

File: src/query/functions-v2/tests/it/scalars/testdata/math.txt [hunk 1]
```diff
@@ -117,28 +117,28 @@ output         : 0.5403023058681398
 ast            : tan(1)
 raw expr       : tan(1_u8)
 checked expr   : tan<Float64>(CAST(1_u8 AS Float64))
-optimized expr : 1.5574077246549023_f64
+optimized expr : 1.557407724654902_f64
```
Comment (leiysky) [inline src/query/functions-v2/tests/it/scalars/testdata/math.txt:120]: Why does the results change?

Comment (andylokandy) [reply src/query/functions-v2/tests/it/scalars/testdata/math.txt:120]: The unit test is not run in one previous PR

**Review by sundy-li — APPROVED** [review]

File: src/query/functions-v2/tests/it/scalars/testdata/math.txt (hunk 1 above)
Comment (andylokandy) [reply src/query/functions-v2/tests/it/scalars/testdata/math.txt:120]: @sundy-li is it difference between running m1 mac and x86 linux?

Comment (sundy-li) [reply src/query/functions-v2/tests/it/scalars/testdata/math.txt]: I do not know, seems the ci works fine with old number.

**Comment by BohuTANG** [comment]
@mergify update

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**